homeassistant/components/sensor/dsmr.py

```diff
@@ -0,0 +1,182 @@
+    devices = []
+
+    # create DSMR interface
+    dsmr = DSMR(hass, config, devices)
+
+    # generate device entities
+    devices += [DSMREntity(name, obis, dsmr) for name, obis in obis_mapping]
+
+    # setup devices
+    yield from hass.loop.create_task(async_add_devices(devices))
+
+    # queue for receiving parsed telegrams from async dsmr reader
+    queue = asyncio.Queue()
+
+    # add asynchronous serial reader/parser task
+    hass.loop.create_task(dsmr.dsmr_parser.read(queue))
+
+    # add task to receive telegrams and update entities
+    hass.loop.create_task(dsmr.read_telegrams(queue))
+
+
+class DSMR:
+    """DSMR interface."""
+
+    def __init__(self, hass, config, devices):
+        """Setup DSMR serial interface, initialize, add device entity list."""
+        from dsmr_parser.serial import (
+            SERIAL_SETTINGS_V4,
+            SERIAL_SETTINGS_V2_2,
+            AsyncSerialReader
+        )
+        from dsmr_parser import telegram_specifications
+
+        # map dsmr version to settings
+        dsmr_versions = {
+            '4': (SERIAL_SETTINGS_V4, telegram_specifications.V4),
+            '2.2': (SERIAL_SETTINGS_V2_2, telegram_specifications.V2_2),
+        }
+
+        # initialize asynchronous telegram reader
+        dsmr_version = config[CONF_DSMR_VERSION]
+        self.dsmr_parser = AsyncSerialReader(
+            device=config[CONF_DEVICE],
+            serial_settings=dsmr_versions[dsmr_version][0],
+            telegram_specification=dsmr_versions[dsmr_version][1],
+        )
+
+        # keep list of device entities to update
+        self.devices = devices
+
+        # initialize empty telegram
+        self._telegram = {}
+
+    @asyncio.coroutine
+    def read_telegrams(self, queue):
+        """Receive parsed telegram from DSMR reader, update entities."""
+        while True:
+            # asynchronously get latest telegram when it arrives
+            self._telegram = yield from queue.get()
+            _LOGGER.debug('received DSMR telegram')
+
+            # make all device entities aware of new telegram
+            for device in self.devices:
+                yield from device.async_update_ha_state()
+
+    @property
+    def telegram(self):
+        """Return telegram object."""
+        return self._telegram
+
+
+class DSMREntity(Entity):
+    """Entity reading values from DSMR telegram."""
+
+    def __init__(self, name, obis, interface):
+        """"Initialize entity."""
+        # human readable name
+        self._name = name
+        # DSMR spec. value identifier
+        self._obis = obis
+        # interface class to get telegram data
+        self._interface = interface
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the sensor, if available."""
+        return getattr(self._interface.telegram.get(self._obis, {}),
```

Comment:
The call itself only reads the value of telegram and does not perform any IO or update sequence. What would be the reason to cache the value in this case?